docs/source/tutorial/10-authenticate-your-queries.mdx

```diff
@@ -0,0 +1,180 @@
+---
+title: "10. Authenticate your queries"
+---
+
+In this section, you will book a flight 🚀! Booking a flight requires being authenticated to the server so the correct person is sent to space! To do that, and since Apollo Android is using [OkHttp](https://square.github.io/okhttp/) to handle HTTP requests, you will leverage the OkHttp [Interceptors](https://square.github.io/okhttp/interceptors/) to add headers to your GraphQL requests.
+
+## Add the interceptor
+
+In `Apollo.kt`, add the `AuthorizationInterceptor` class:
+
+```kotlin:title=app/src/main/kotlin/com/example/rocketreserver/Apollo.kt
+private class AuthorizationInterceptor(val context: Context): Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request().newBuilder()
+            .addHeader("Authorization", User.getToken(context) ?: "")
+            .build()
+
+        return chain.proceed(request)
+    }
+}
+```
+
+This interceptor will append a `"Authorization: $token"` HTTP header to every request.
+
+## Use the interceptor
+
+Create a custom OkHttpClient that will use this interceptor and pass it to the ApolloClient:
+
+```kotlin:title=app/src/main/kotlin/com/example/rocketreserver/Apollo.kt
+    instance = ApolloClient.builder()
+        .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/")
+        .okHttpClient(OkHttpClient.Builder()
+            .addInterceptor(AuthorizationInterceptor(context))
+            .build()
+        )
+        .build()
+```
+
+Notice that `Apollo.kt` now requires a context to create the interceptor and read the token from the `EncryptedSharedPreferences`. `apolloClient` cannot be a top level variable anymore so transform it to a top-level method that takes a `context` parameter instead:
```

Comment:
I realized that you also use `requireContext` to get apolloClient in earlier tutorials but `context` is introduced just now. Is that something to fix?

Reply:
Yup, my mistake, I copy/pasted code from the final state too early in the tutorial. I fixed one instance in step 8, "add a details view". Were there other instances?

Reply:
I cannot see them anymore. All good.